### PR TITLE
fix edition of effort curve

### LINKED ISSUE #6579: Unable to edit speed effort curve
### What happened?

In the rolling stock editor, when adding a new electric profile or a new power restriction code on a duplicated rolling stock, it "breaks" the whole related traction mode. This leads to the impossibility to edit any speed effort curve of any electrical profile / power restriction of the traction mode.
This issue doesn't happen on a rolling stock created from scratch.

### What did you expect to happen?

Adding a new electrical profile or power restriction code shouldn't break anything.

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to rolling stock editor
2. Select a rollling stock with power restrictions (a tgv for example), edit it and go to speed effort curve tab
3. Check that you can edit any curve
4. Now add a power restriction code in any traction mode / electrical profile and try to edit any curve

### What operating system, browser and environment are you using?

- **Browser**: Firefox
- **OS**: Ubuntu 
- **Env**: Dev

### OSRD version (top right corner `Account` button > `Informations`)

dev 5895b87

### PR DESCRIPTION
**Pull request by kmer2016**
close #6579

### CODE REVIEW

**Review by SharglutDev — APPROVED** [review]
Thank you for this PR and good job to spot this sneaky bug, everything works great now ! Lgtm :)

**Review by SarahBellaha — APPROVED** [review]
LGTM 💪

File: front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorCurves.tsx [hunk 1]
```diff
@@ -200,39 +208,43 @@ const RollingStockEditorCurves = ({
...
-                    ? selectedParams.powerRestriction
-                    : selectedParams.electricalProfile
-                }
-              />
-            )}
-            {children}
+      {selectedTractionMode &&
+        selectedCurve &&
+        selectedCurveIndex !== null &&
+        selectedTractionModeCurves !== null && (
```
Comment (clarani) [inline front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorCurves.tsx]: You don't need the null here
```suggestion
        selectedTractionModeCurves && (
```

**Review by clarani — APPROVED** [review]
LGTM, tested :white_check_mark:

**Pull request merged**